### ISSUE TITLE
Contribution: gridcoinresearchd BASH completion

### ISSUE THREAD

**Issue by smoe** [gridcoin-community/Gridcoin-Research#1410]
It may be beneficial for users to have the BASH command line completion support for gridcoinresearchd.  This gist
https://gist.github.com/smoe/0cfa4ad8982b55f1ff30c40ad1593bc0
has an implementation.

**Comment by RoboticMind** [comment]
Feel like this should probably go somewhere else besides being in a GitHub issue. Not sure where exactly it would fit best but probably not in an issue here.

**Comment by smoe** [comment]
Thank you for your support. I had already forgotten about it. Maybe someone could ping the individual providing the Debian package about it? With a bit of look someone knows where this fits and transforms it into a pull request.

**Comment by RoboticMind** [comment]
@caraka is the builder of the Debian package. @caraka See the messages above for more information

**Comment by caraka** [comment]
Thanks @smoe I hadn't seen this. IIRC, it needs a separate 'package' that gets pulled in when gridcoinresearchd is installed. It won't be immediate, but I will look into how this is normally done. Thanks for your contribution! I tire kicked this idea about 4 years ago when I first started building, and put it into the too hard bin in part because our RPC stuff was such a mess. With our recently purged/cleaned RPC commands, this is timely.

**Comment by smoe** [comment]
This is nice to hear, @caraka . Many thanks. This was my very first shot at bash completions, so there may be a point to have this reviewed by someone more experienced. I don't think this merrits a separate package, really. I propose upstream (you?) to maintain it somewhere in the gridcoinresearchd's source tree and the Ubuntu/SuSE/RedHat/Centos/... packaging places that file in /etc/bash-completion.d/ .

**Comment by smoe** [comment]
Btw, I am one of the co-maintainers of the Debian BOINC package (https://salsa.debian.org/pkg-boinc-team/boinc). If you want something sponsored or if there is something else that Debian should do for you - please tell me. Also, having no idea if you currently have a repository for the Ubuntu packaging somewhere - you are most cordially invited to join the https://salsa.debian.org/pkg-boinc-team.

**Comment by RoboticMind** [comment]
Maybe this could go in the contrib folder somewhere?

**Comment by jamescowens** [comment]
Yes. And this file simply needs to be placed in /etc/bash.completion.d and logout login (to source the file) for it to become active. We ought to be able to do this in the normal packages and the install command of make.

**Comment by jamescowens** [comment]
@caraka I have assigned this to you for the Debian/Ubuntu builds. I don't think this requires a separate package. We probably should detect whether the standard bash-completion package is already installed as a precedent.

**Comment by jamescowens** [comment]
@caraka, this seems pretty simple to put in the packaging... can we get this done for Ingrid?

**Comment by smoe** [comment]
> 
> 
> @caraka I have assigned this to you for the Debian/Ubuntu builds. I don't think this requires a separate package. We probably should detect whether the standard bash-completion package is already installed as a precedent.

The directory /etc/bash_completion.d can exist independently from the bash-completion package. 
```
$ apt-file search /etc/bash_completion.d/
cdcd: /etc/bash_completion.d/cdcd
dlocate: /etc/bash_completion.d/dlocate-completion
git: /etc/bash_completion.d/git-prompt
git-flow: /etc/bash_completion.d/git-flow
gmic: /etc/bash_completion.d/gmic
lxctl: /etc/bash_completion.d/lxctl
...
lxctl: /etc/bash_completion.d/lxctl_vz2lxc
qtltools: /etc/bash_completion.d/QTLtools_bash_autocomplete.bash
rosbash: /etc/bash_completion.d/rosbash
svn-buildpackage: /etc/bash_completion.d/svn-buildpackage
```
I just wished someone with more experience on these bash-completions would give that patch a peer review.

**Comment by jamescowens** [comment]
I am moving this to Janice. It is a nice to have, and we can push it to the next milestone.

**Comment by jamescowens** [comment]
@caraka ping on this. This is independent enough that this can still go in the Kermit's Mom release.
